File: be/src/projects/dto/create-project.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsString, IsOptional, IsDateString, IsUUID, IsEnum } from 'class-validator';

export class CreateProjectDto {
  @ApiProperty({ example: 'Tên dự án' })
  @IsString()
  name: string;

  @ApiProperty({ example: 'Mô tả dự án' })
  @IsOptional()
  @IsString()
  description?: string;

  @ApiProperty({ example: '2025-01-01' })
  @IsOptional()
  @IsDateString()
  deadline?: string;

  @ApiProperty({ example: 'todo' })
  @IsOptional()
  @IsEnum(['todo', 'doing', 'done'])
  status?: 'todo' | 'doing' | 'done';

  @ApiProperty({ example: '123e4567-e89b-12d3-a456-426614174000' })
  @IsOptional()
  @IsUUID()
  groupId?: string;
}

File: backend-java/src/main/java/com/qlda/backend/projects/dto/CreateProjectDto.java
package com.qlda.backend.projects.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.Data;

import java.time.LocalDate;

@Data
public class CreateProjectDto {
    @NotBlank(message = "Tên dự án không được để trống")
    private String name;

    private String description;
    private LocalDate startDate;
    private LocalDate deadline;

    @Pattern(regexp = "todo|doing|done", message = "Status phải là todo, doing hoặc done")
    private String status = "todo";

    private String groupId;
}

